content/articles/switch-vs-router-vs-hub/index.md

```diff
@@ -0,0 +1,117 @@
+- They are expensive compared to other devices in a network.
+- They need to analyze data making them slower.
+- Low bandwidth because of dynamic router communication.
+
+Let's look at the differences of each of them:
+
+### Layer of each component in the OSI model
+The [OSI Model](https://www.imperva.com/learn/application-security/osi-model/) (Open Systems Interconnection Model) is a 7 layer model that is used to describe in a pictorial way how computer systems communicate. Switch, Router, and Hub each operate on a different layer.
+
+#### Switch
+A switch is located on the OSI model's Data Link layer i.e the second layer. The link layer is specific to the medium over which the packet is traveling such as Ethernet and the Mac Address are part of this layer.
+
+#### Router
+A router works in the Network Layer of the OSI model and it is the third layer.
+
+#### Hub
+Hub works in the Physical Layer of the OSI model and it is the first layer.
+
+### Function of each Device
+#### Switch
+It allows various connections of many devices in the same network and the management of port and VLAN security settings.
+
+#### Router
+Its major purpose is to connect many types of networks at the same time using adaptive and non-adaptive routing. 
+
+#### hub
+Hub is a simple and cheap networking device that allows a bunch of computers to be connected to a single network 
+
+### Application of each Device
+#### Switch
+It is commonly used in local area networks for connecting many nodes.
+
+#### Souter
+It is commonly used in Local Area Network and Metropolitan Area Network(MAN).
+
+#### hub
+It is similar to Switch because it is used in the local area network(LAN).
+
+### Mode of data transmission
+It defines the direction in which data flow between two communicating devices. There are three types of transmission modes. Read this [documentation](https://afteracademy.com/blog/what-are-the-data-transmission-modes-in-a-network) for more information on the Mode of data transmission
+
+#### Switch
+It supports full-duplex transmission thus a bunch of computers can send data at the same time.
+
+#### Router
+It supports full-duplex transmission thus a bunch of computers can send data at the same time.
+
+#### Hub
+It supports half duplex in transmission thus only one node can send data at a time.
+
+### Adresses used in each devives
+#### Switch
+Switch stores and uses the MAC address of a device to transfer data.
+
+#### Router
+A router uses the IP address to transfer data between networks.
+
+#### Hub
+Hub does not store any MAC/IP address to transfer data.
+
+### Transmission of Data
+#### Switch
+Switch transmit data from one device to another in form of [frames](https://en.wikipedia.org/wiki/Frame_(networking)).
+
+#### Router 
+The router transmits data from one network to another in form of [packets](https://www.cloudflare.com/learning/network-layer/what-is-a-packet/).
+
+#### Hub
+Hub transmits data from one device to another in form of binary bits.
+

```

Comment:
Finally, you could add a summary or conclusion section to summarize the points in your article. It's always good practice to add that section.